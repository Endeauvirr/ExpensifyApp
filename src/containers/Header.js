import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <ul>
      <li>
        <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Main page</NavLink>
      </li>
      <li>
        <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
      </li>
      <li>
        <button onClick={startLogout}>Log out</button>
      </li>
    </ul>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
