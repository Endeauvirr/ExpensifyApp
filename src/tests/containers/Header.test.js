import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../containers/Header';

// Snapshot testing , snapshoty pozwalają na badanie jak zmieniają się dane w czasie

test('Should render Header container correctly', () => {
  // ==== Using react-test-renderer

  // const renderer = new ReactShallowRenderer();

  // renderer.render(<Header />);

  // expect(renderer.getRenderOutput()).toMatchSnapshot();


  // ==== Using Enzyme

  const wrapper = shallow(<Header />);

  expect(wrapper).toMatchSnapshot();
});

