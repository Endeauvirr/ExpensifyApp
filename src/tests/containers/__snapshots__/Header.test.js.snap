// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header container correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Main page
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/create"
      >
        Create expense
      </NavLink>
    </li>
  </ul>
</header>
`;
